test_plans/bfloat16.asciidoc

```diff
@@ -0,0 +1,118 @@
+:sectnums:
+:xrefstyle: short
+
+= Test plan for bfloat16
+
+This is a test plan for the APIs described in
+https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_oneapi_bfloat16.asciidoc[sycl_ext_oneapi_bfloat16]
+
+== Testing scope
+
+=== Device coverage
+
+All of the tests described below are performed only on the default device that
+is selected on the CTS command line.
+
+=== Type coverage
+
+All of the tests described below are performed using each of the following `typename T`:
+
+* `char`
+* `short`
+* `int`
+* `long long`
+* `std::size_t`
+* `bool`
+* `float`
+
+In addition, if the device has aspect `aspect::fp64`:
+
+* `double`
+
+In addition, if the device has `aspect::fp16`:
+
+* `sycl::half`
+
+Note that `T` must be convertible to `float`
+
+== Tests
+
+=== Constructors
+
+`bfloat16()` +
+`bfloat16(const bfloat16 &)`
+
+Check that:
+
+* `std::is_default_constructible_v<bfloat16> == true`.
+* `std::is_copy_constructible_v<bfloat16> == true`.
+
+`bfloat16(const float &a)` +
+`bfloat16(const sycl::half &a)`
+
+Since `bfloat16` has 7 bits significand, epsilon value for this type is 2^-7, or 0.0078125. Therefore, objects created using these constructors must be in the range `(a - 0.0078125, a + 0.0078125)`. Create `bfloat16` object named `bf` and verify:
+
+* `sizeof(bf) == 2`
+* `(bf > a - epsilon) && (bf < a + epsilon)`
```

Comment:
```suggestion
* `(bf > bf - epsilon) && (bf < bf + epsilon)`
```

I assume this is what you meant?  Otherwise, I don't understand what `a` is.

Reply:
> I assume this is what you meant? Otherwise, I don't understand what `a` is.

I am unsure.

Reply:
`a` is the argument of `bfloat16` constructors listed above (`bfloat16(const float &a)`and 
`bfloat16(const sycl::half &a)`)

Reply:
Oh, I see.  Can you clarify this by saying:

```suggestion
* `(bf > a - epsilon) && (bf < a + epsilon)`, where `a` is the value passed to the constructor.
```

Reply:
Applied.